Apps/W1/HybridGP/app/src/Migration/History/GPPopulateHistTables.Codeunit.al

```diff
@@ -0,0 +1,1218 @@
+#pragma warning restore AA0139
+
+                HistGenJournalLine."Journal Entry No." := Format(GPGL20000.JRNENTRY);
+                HistGenJournalLine."Account No." := CopyStr(Format(GPGL00105.ACTNUMST), 1, MaxStrLen(HistGenJournalLine."Account No."));
+                HistGenJournalLine."Sequence No." := GPGL20000.SEQNUMBR;
+                HistGenJournalLine."Audit Code" := GPGL20000.TRXSORCE;
+                HistGenJournalLine.Closed := false;
+                HistGenJournalLine.Year := GPGL20000.OPENYEAR;
+                HistGenJournalLine."Date" := DT2Date(OutlookSynchTypeConv.LocalDT2UTC(GPGL20000.TRXDATE));
+                HistGenJournalLine."Currency Code" := CopyStr(GPGL20000.CURNCYID, 1, MaxStrLen(HistGenJournalLine."Currency Code"));
+                HistGenJournalLine."Debit Amount" := GPGL20000.DEBITAMT;
+                HistGenJournalLine."Orig. Debit Amount" := GPGL20000.ORDBTAMT;
+                HistGenJournalLine."Credit Amount" := GPGL20000.CRDTAMNT;
+                HistGenJournalLine."Orig. Credit Amount" := GPGL20000.ORCRDAMT;
+                HistGenJournalLine."Document Type" := GPGL20000.SOURCDOC;
+                HistGenJournalLine."Reference Desc." := GPGL20000.REFRENCE;
+                HistGenJournalLine."Description" := GPGL20000.DSCRIPTN;
+                HistGenJournalLine."Orig. Trx. Source No." := GPGL20000.ORTRXSRC;
+                HistGenJournalLine."User" := GPGL20000.USWHPSTD;
+                HistGenJournalLine."Custom1" := GPGL20000.User_Defined_Text01;
+                HistGenJournalLine."Custom2" := GPGL20000.User_Defined_Text02;
+
+                if GPGL20000.SERIES <> GPPayrollSeriesId() then begin
+#pragma warning disable AA0139
+                    HistGenJournalLine."Orig. Document No." := GPGL20000.ORDOCNUM.TrimEnd();
+#pragma warning restore AA0139
+                    HistGenJournalLine."Source No." := GPGL20000.ORMSTRID;
+                    HistGenJournalLine."Source Name" := CopyStr(GPGL20000.ORMSTRNM, 1, MaxStrLen(HistGenJournalLine."Source Name"));
+                end;
+
+                if HistGenJournalLine.Insert() then
+                    PopulateGLOpenYearItemTransaction(GPGL20000)
+                else
+                    HistMigrationStatusMgmt.ReportLastError("Hist. Migration Step Type"::"GP GL Journal Trx.", GPGL20000.TRXSORCE, true);
+
+                AfterProcessedNextRecord();
+            end;
+        until GPGL20000.Next() = 0;
+    end;
+
+    local procedure PopulateHistoricalYearPostedTransactions()
+    var
+        GPGL30000: Record "GP GL30000";
+        GPGL00105: Record "GP GL00105";
+        HistGenJournalLine: Record "Hist. Gen. Journal Line";
+        OutlookSynchTypeConv: Codeunit "Outlook Synch. Type Conv";
+        InitialHistYear: Integer;
+    begin
+        InitialHistYear := GPCompanyAdditionalSettings.GetHistInitialYear();
+        if InitialHistYear > 0 then
+            GPGL30000.SetFilter(HSTYEAR, '>=%1', InitialHistYear);
```

Comment:
This applies for the other tables in this codeunit. Please ensure that we have indexes (keys) on all filters that are used if the table can contain millions of records. It will speed the things up.

The important part is that all fields filtered on should be reflected in the index. 
e.g. if the table below is large, we would need a key for both fields:
        GPSOPTrxAmountsHist.SetRange(SOPTYPE, GPSOPTrxHist.SOPTYPE);
        GPSOPTrxAmountsHist.SetRange(SOPNUMBE, GPSOPTrxHist.SOPNUMBE);
        
        key(SOPTYPE,SOPNUMBE)